### PR TITLE
agent: Include autoscaling bounds annotation in per vm metrics

### PR DESCRIPTION
**Pull request by shayanh**
Fixes https://github.com/neondatabase/autoscaling/issues/686

### CODE REVIEW

File: pkg/agent/watch.go [hunk 1]
```diff
@@ -202,6 +204,18 @@ func (e vmEvent) Format(state fmt.State, verb rune) {
 	}
 }
 
+func extractAutoScalingBounds(vm *vmapi.VirtualMachine) *api.ScalingBounds {
```
Comment (sharnoff) [inline pkg/agent/watch.go:207]:
```suggestion
func extractAutoscalingBounds(vm *vmapi.VirtualMachine) *api.ScalingBounds {
```

Comment (sharnoff) [inline pkg/agent/watch.go:207]: It'd be good to have a comment here explaining what it does and why we have this even though `api.ExtractVmInfo` exists and does some of it

File: pkg/agent/watch.go [hunk 2]
```diff
@@ -230,11 +245,24 @@ func makeVMCPUMetrics(vm *vmapi.VirtualMachine) []vmMetric {
...
 	}
 
+	// metrics from autoscaling bounds annotation
+	if bounds := extractAutoScalingBounds(vm); bounds != nil {
+		boundPairs := []pair[vmResourceValueType, resource.Quantity]{
+			{vmResourceValueAutoscalingMin, bounds.Min.CPU},
+			{vmResourceValueAutoscalingMax, bounds.Max.CPU},
+		}
+		for _, p := range boundPairs {
+			m := makeVMMetric(vm, p.first, vmapi.MilliCPUFromResourceQuantity(p.second).AsFloat64())
```
Comment (sharnoff) [inline pkg/agent/watch.go:261]: worth explaining why we're converting to `vmapi.MilliCPU` first instead of directly converting `resource.Quantity` to float

**Review by sharnoff — COMMENTED** [review]
generally looks good

File: pkg/agent/watch.go (hunk 2 above)
Comment (shayanh) [reply pkg/agent/watch.go:261]: I wrote it this way to keep it consistent with the other CPU metrics. Do you have another reason in mind?

Comment (sharnoff) [reply pkg/agent/watch.go:265]: oh, I think that's how it should be - I'm just saying that there should be a comment explaining why to do it this way (because `resource.Quantity.AsApproximateFloat64()` is quite inaccurate)

Comment (sharnoff) [reply pkg/agent/watch.go:265]: https://github.com/neondatabase/autoscaling/blob/90e79ff655aac6774fa806e7928c4a3f68bbd273/neonvm/apis/neonvm/v1/virtualmachine_types.go#L200-L206

Comment (shayanh) [reply pkg/agent/watch.go:266]: fixed in 4ff769668e14024cee2b8679e79a7cd12c149664

File: pkg/agent/watch.go [hunk 3]
```diff
@@ -202,6 +204,22 @@ func (e vmEvent) Format(state fmt.State, verb rune) {
 	}
 }
 
+// extractAutoscalingBounds extracts the ScalingBounds from a VM's autoscaling
+// annotation. We have defined this function for the purpose of metrics, so we
+// could track scaling bounds changes on a VM. api.ExtractVmInfo doesn't provide
+// this data, since its CPU and memory values might come from vm.Spec.
```
Comment (sharnoff) [inline pkg/agent/watch.go:210]:
```suggestion
// extractAutoscalingBounds extracts the ScalingBounds from a VM's autoscaling
// annotation, for the purpose of exposing it in per-VM metrics.
//
// We're not reusing api.ExtractVmInfo even though it also looks at the bounds
// annotation, because its data is less precise - CPU and memory values might
// come from the VM spec without us knowing.
```

I'd reword + simplify + add a line break.

I found it hard to parse the original because logically distinct thoughts started mid-line, and felt there wasn't a clear connection of why we might want to use `api.ExtractVmInfo` in the first place.
IIUC, the bit about "so we could track scaling bound changes on a VM" isn't _quite_ accurate, because in theory that data is fully available in the control plane database - it's just easier to provide it all in one place.

Feel free to reword to fit your style — figured it'd be easier to show what I meant, rather than tell :)

**Review by sharnoff — APPROVED** [review]
one comment nit. feel free to merge once changed
